crates/client/eth/src/config.rs

```diff
@@ -0,0 +1,77 @@
+use alloy::primitives::Address;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::path::PathBuf;
+
+use crate::error::Error;
+
+/// Default Anvil local endpoint
+pub const DEFAULT_RPC_ENDPOINT: &str = "http://127.0.0.1:8545";
+/// Default Anvil chain ID
+pub const DEFAULT_CHAIN_ID: u64 = 31337;
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct EthereumClientConfig {
+    #[serde(default)]
+    pub provider: EthereumProviderConfig,
+    #[serde(default)]
+    pub contracts: StarknetContracts,
```

Comment:
I would replace this field with 
`pub core_contrac_address: Eth::ContractAddress`, and don't add `    #[serde(default)]`.


We don't want at all to use the default string `""` if this is not given by the config. The deserialization should fail instead

Reply:
here you mean `alloy::primitives::Address` or `starknet_core::types::EthAddress`?